src/main/java/edu/ucla/library/libcal/services/OAuthTokenServiceImpl.java

```diff
@@ -0,0 +1,108 @@
+
+package edu.ucla.library.libcal.services;
+
+import edu.ucla.library.libcal.Config;
+import edu.ucla.library.libcal.Constants;
+import edu.ucla.library.libcal.JsonKeys;
+import edu.ucla.library.libcal.MessageCodes;
+
+import info.freelibrary.util.Logger;
+import info.freelibrary.util.LoggerFactory;
+
+import io.vertx.core.Future;
+import io.vertx.core.Vertx;
+import io.vertx.core.json.JsonObject;
+import io.vertx.ext.auth.User;
+import io.vertx.ext.auth.oauth2.OAuth2Auth;
+
+/**
+ * Implementation of {@link OAuthTokenService}.
+ * <p>
+ * FIXME: the likelihood of {@link OAuthTokenService#getBearerToken(Vertx)} returning an expired token is probably too
+ * high for a production service.
+ */
+public class OAuthTokenServiceImpl implements OAuthTokenService {
+
+    /**
+     * The logger.
+     */
+    private static final Logger LOGGER = LoggerFactory.getLogger(OAuthTokenServiceImpl.class, MessageCodes.BUNDLE);
+
+    /**
+     * The Vert.x instance.
+     */
+    private final Vertx myVertx;
+
+    /**
+     * The ID of the periodic timer for checking if the OAuth token needs refreshing.
+     */
+    private final long myTimerId;
+
+    /**
+     * The OAuth authentication provider.
+     */
+    private final OAuth2Auth myAuthProvider;
+
+    /**
+     * LibCal credentials represented as an OAuth token.
+     */
+    private User myToken;
+
+    /**
+     * Creates an instance of the service. Use {@link OAuthTokenService#create(Vertx, JsonObject)} to invoke.
+     *
+     * @param aVertx A Vert.x instance
+     * @param aConfig A configuration
+     * @param anAuthProvider An OAuth authentication provider
+     * @param aToken A token that was returned by authenticating against {@code anAuthProvider}
+     */
+    public OAuthTokenServiceImpl(final Vertx aVertx, final JsonObject aConfig, final OAuth2Auth anAuthProvider,
```

Comment:
Ha! I've wanted to use "an" many times but mistakenly thought it wouldn't pass the naming rule check (or that I wasn't being true to the check or something). Welp, now I need to go update all my code that uses "a" with words that start with a vowel. It definitely reads better.